code/modules/spells/spell_types/summonitem.dm

```diff
@@ -79,12 +78,15 @@
 						M.dropItemToGround(item_to_retrieve)
 
 					else
-						if(istype(item_to_retrieve.loc, /obj/machinery/portable_atmospherics/)) //Edge cases for moved machinery
-							var/obj/machinery/portable_atmospherics/P = item_to_retrieve.loc
+						var/obj/retrieved_item = item_to_retrieve.loc
+						if(retrieved_item.anchored)
+							break
```

Comment:
just return, don't break, the way they've done this of embedding this handling inside the for loop is unfortunate

Reply:
If it returns then it wouldn't get the item at all, I just didn't want it to take the whole machine its in, with it

Reply:
I'm not really sure what you mean by this, it only checks the loc once, there's no loop here?